src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -126,14 +128,23 @@ public static BugsnagEventRequest makeEventDataRequest(String eventDataRequestId
             "organizations", BUGSNAG_ORGANIZATION, "event_data_requests", eventDataRequestId
         );
         LOG.debug("Making Bugsnag request: {}", eventDataRequestUri);
-        HttpResponse<String> response = HttpUtils.httpRequestRawResponse(
-            eventDataRequestUri,
-            CONNECTION_TIMEOUT_IN_SECONDS,
-            create ? HttpMethod.POST : HttpMethod.GET,
-            BUGSNAG_HEADERS,
-            create ? EVENT_REQUEST_FILTER : null
-        );
-        return JsonUtils.getPOJOFromHttpBody(response, BugsnagEventRequest.class);
+
+        try {
+            HttpResponse response = HttpUtils.httpRequestRawResponse(
+                eventDataRequestUri,
+                CONNECTION_TIMEOUT_IN_SECONDS,
+                create ? HttpMethod.POST : HttpMethod.GET,
+                BUGSNAG_HEADERS,
+                create ? EVENT_REQUEST_FILTER : null,
+                true
+            );
+            return JsonUtils.getPOJOFromHttpBody(response, BugsnagEventRequest.class);
+        } catch (HttpTimeoutException e) {
+            LOG.error("Request to {} timed out after {} seconds.", eventDataRequestUri, CONNECTION_TIMEOUT_IN_SECONDS, e);
+        } catch (IOException e) {
+            BugsnagReporter.reportErrorToBugsnag("Error requesting data from URI", eventDataRequestUri, e);
+        }
```

Comment:
Any reason not to have the `httpRequestRawResponse` handle the IOException similar to Data Tools:
https://github.com/ibi-group/datatools-server/blob/6829065126adbd7689614ddf5d52b5efb736ed86/src/main/java/com/conveyal/datatools/manager/utils/HttpUtils.java#L82-L88

It looks like we're reporting the Bugsnag error in multiple places, when maybe we should just do this once in the util class. I know we've gone back and forth a bit on utils like this (e.g. JSON parsing), so I'm just raising this to make sure we have a consistent approach.

Reply:
@landonreed this also raises another issue, we should probably be using the CloseableHttpClient and try with resources to guarantee the connection is closed. However, this will have an impact on calling services. As the connection will be closed the response body would need to be extracted here and provided with the status code in a bespoke class (probably using a Response Handler).

The thinking behind reporting Bugsnag errors in multiple places was to log any time outs, but not to report them to Bugsnag. Perhaps this is better off in HttpUtils and is used across the board?

Reply:
Yep, that all sounds good:
- using closeablehttpclient in try/catch with resources
- adding new class for response handler
- handle logging timeouts/bugsnag logging in httputils

Reply:
This approach has now been implemented.